pubsub/consumer.go

```diff
@@ -90,6 +99,18 @@ func (c *Consumer[Request, Response]) heartBeatKey() string {
 	return heartBeatKey(c.id)
 }
 
+// deleteHeartBeat deletes the heartbeat to indicate it is being shut down.
+func (c *Consumer[Request, Response]) deleteHeartBeat(ctx context.Context) {
+	c.terminating.Store(true)
```

Comment:
This can still race with the heartbeat in a way that hearbeat will be written right after it's deleted.
There are two immediate solutions I can think of:
1) Use a mutex for checking terminating + send redis request (which I don't like because it sends a remote request inside a mutex
2) call deleteHeartBeat After c.StopWaiter.StopAndWait(). Stop and Wait.. waits for all threads to terminate before it returns so no race.

Reply:
Done.